state/sqlserver/metadata.go

```diff
@@ -0,0 +1,298 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqlserver
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+	"unicode"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	mssql "github.com/microsoft/go-mssqldb"
+	"github.com/microsoft/go-mssqldb/msdsn"
+
+	"github.com/dapr/components-contrib/internal/authentication/azure"
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/kit/ptr"
+)
+
+const (
+	connectionStringKey  = "connectionString"
+	tableNameKey         = "tableName"
+	metadataTableNameKey = "metadataTableName"
+	schemaKey            = "schema"
+	keyTypeKey           = "keyType"
+	keyLengthKey         = "keyLength"
+	indexedPropertiesKey = "indexedProperties"
+	keyColumnName        = "Key"
+	rowVersionColumnName = "RowVersion"
+	databaseNameKey      = "databaseName"
+	cleanupIntervalKey   = "cleanupIntervalInSeconds"
+
+	defaultKeyLength       = 200
+	defaultSchema          = "dbo"
+	defaultDatabase        = "dapr"
+	defaultTable           = "state"
+	defaultMetaTable       = "dapr_metadata"
```

Comment:
These are some common default names across other components as well, like postgresql etc.
Should we rather make these constants parts of common constants file, like in https://github.com/dapr/components-contrib/tree/master/state/utils ?

Reply:
For these ones specifically, my preference is on not doing that because each component can choose a different implementation.